core/src/test/java/org/infinispan/distribution/rehash/NonTxPutIfAbsentDuringJoinStressTest.java

```diff
@@ -45,40 +47,35 @@ protected void createCacheManagers() throws Throwable {
    }
 
    private ConfigurationBuilder getConfigurationBuilder() {
-      ConfigurationBuilder c = new ConfigurationBuilder();
-      c.clustering().cacheMode(CacheMode.DIST_SYNC);
-      c.transaction().transactionMode(TransactionMode.NON_TRANSACTIONAL);
-      return c;
+      return getDefaultClusteredCacheConfig(cacheMode, false);
    }
 
-   @Test(groups = "unstable", description = "See ISPN-3918")
+// The test is unstable only in dist/repl cache, scattered cache is fine
```

Comment:
hmmm would it be possible to fix this test somehow? I'm slightly afraid that introducing this feature we will blur the solution a little bit.